projects/packages/forms/src/blocks/contact-form/components/jetpack-integrations-modal/mailpoet-card.tsx

```diff
@@ -24,12 +37,51 @@ const MailPoetCard = ( {
 		marketingUrl = '',
 	} = data || {};
 
+	const mailpoetLists: MailPoetList[] = useMemo(
+		() => ( Array.isArray( data?.details?.lists ) ? ( data.details.lists as MailPoetList[] ) : [] ),
+		[ data?.details?.lists ]
+	);
+
+	useEffect( () => {
+		if ( ! mailpoet.enabledForForm ) {
+			return;
+		}
+
+		// If there are no lists, clear the selection
+		if ( mailpoetLists.length === 0 ) {
+			if ( mailpoet.listId || mailpoet.listName ) {
+				setAttributes( {
+					mailpoet: {
+						...mailpoet,
+						listId: null,
+						listName: null,
+					},
+				} );
+			}
+			return;
+		}
+
+		// If no list is selected, or the selected list no longer exists, set to the first available
+		const listIsValid =
+			mailpoet.listId && mailpoetLists.some( list => list.id === mailpoet.listId );
+		if ( ! listIsValid ) {
+			setAttributes( {
+				mailpoet: {
+					...mailpoet,
+					listId: mailpoetLists[ 0 ].id,
+					listName: mailpoetLists[ 0 ].name,
+				},
+			} );
+		}
+	}, [ mailpoet, mailpoetLists, setAttributes ] );
```

Comment:
The useEffect above does a few things:
* Ensures the MailPoet listID is properly set even if a user never adjusts it. Otherwise the block editor will not set this if the user just wants to use the default selected list and doesn't actively change it. 
* Validate a list ID. I ran into issue if you delete a list that was previously selected, it will still be set here. But we need to clear that.